addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/httputils/HtmlContextAnalyserUnitTest.java

```diff
@@ -0,0 +1,160 @@
+
+    @Test
+    void shouldGetParsedAttributes() {
+        String attr = " onclick = \"alert(100)\" accesskey = \"x\" n=\"\"";
+        Map<String, String> attrMap = HtmlContextAnalyser.parseAttributes(attr);
+        assertThat(attrMap.get("onclick"), is(equalTo("alert(100)")));
+        assertThat(attrMap.get("accesskey"), is(equalTo("x")));
+        assertThat(attrMap.get("n"), is(equalTo("")));
+    }
+
+    @Test
+    void shouldGetParsedTags() {
+        String tag = "<body onclick = \"alert(100)\" accesskey = \"x\" n=\"\">";
+        Map<String, Map<String, String>> tagMap = HtmlContextAnalyser.parseTag(tag);
+        Map<String, String> attrMap = tagMap.get("body");
+        assertThat(attrMap.get("onclick"), is(equalTo("alert(100)")));
+        assertThat(tagMap.get("body").get("accesskey"), is(equalTo("x")));
+    }
+
+    @Test
+    void shouldParseTagsWithoutCompletion() {
+        String tag = "<html> <body> <p accesskey='x' onclick='alert(100)'> Hello </html>";
+        Map<String, Map<String, String>> tagMap = HtmlContextAnalyser.parseTag(tag);
+        Map<String, String> attrMap = tagMap.get("p");
+        assertThat(attrMap.get("onclick"), is(equalTo("alert(100)")));
+        assertThat(tagMap.get("p").get("accesskey"), is(equalTo("x")));
+    }
+
+    @Test
+    void shouldNotRaiseNullPointerException() {
+        String tag = "<html> <body> <p accesskey='x' onclick='alert(100)' Hello </html>";
+        Map<String, Map<String, String>> tagMap = HtmlContextAnalyser.parseTag(tag);
+        assertThat(tagMap.size(), is(equalTo(3)));
+        assertThat(tagMap.get("html").size(), is(equalTo(0)));
+        assertThat(tagMap.get("p").size(), is(equalTo(2)));
+        assertThat(tagMap.get("p").get("Hello"), is(equalTo(null)));
+    }
+
+    @Test
+    void shouldNotParseComments() {
+        String tag =
+                "<html> <body> <p accesskey='x' onclick='alert(100)'> Hello <!-- comment --> </html>";
+        Map<String, Map<String, String>> tagMap = HtmlContextAnalyser.parseTag(tag);
+        assertThat(tagMap.size(), is(equalTo(3)));
+        assertThat(tagMap.get("html").size(), is(equalTo(0)));
+        assertThat(tagMap.get("p").size(), is(equalTo(2)));
+        assertThat(tagMap.get("p").get("Hello"), is(equalTo(null)));
+    }
+
+    @Test
+    void shouldParseAttributesWithoutQuotesForValues() {
+        String attr = " onclick = alert(100) accesskey = x n=\"\" bac = ''";
+        Map<String, String> attrMap = HtmlContextAnalyser.parseAttributes(attr);
+        assertThat(attrMap.get("onclick"), is(equalTo("alert(100)")));
+        assertThat(attrMap.get("accesskey"), is(equalTo("x")));
+        assertThat(attrMap.get("n"), is(equalTo("")));
+        assertThat(attrMap.get("bac"), is(equalTo("")));
+    }
+
+    @Test
+    void shouldWorkWhenJerichoDoesnt() {
+        // parse using jericho first.. as it is being parsed in HtmlContextAnalyser Code
+        String tag = "<html><body><0W45pz4p is 1337></body></html>";
```

Comment:
This is properly parsed, the elements are just not exposed directly: https://javadoc.io/doc/net.htmlparser.jericho/jericho-html/latest/net/htmlparser/jericho/Tag.html#isUnregistered() (Doing `.replace("0W45pz4p", "OWASPZAP")` will no longer consider them unregistered and are returned in the usual methods.)

Reply:
But when will I know that using the analyser that I should be going for replacing it and with "what"?. Like the eyecatcher here is parsed as an unregistered tag and can be accessed with the index number from the source. But is replacing the eyecatcher a good idea? Or maybe passing a new eyecatcher.....

Reply:
That should be part of the analyser itself, it could replace always. In the analyses phase it does not really matter what the eye catcher is, as long as we keep the same length to match the existing indices. Even if we access the unregistered element directly the attributes are still not available.

Reply:
Like here 
```
if (StringUtils.strip(element.getContent().toString()).contains(target)) {
                    Map<String, Map<String, String>> tagMap =
                            parseTag(StringUtils.strip(element.getContent().toString()));
                    for (String tagName : tagMap.keySet()) {
                        if (target.contains(tagName)) {
                            context.setInElementName(true);
                            context.setElementName(tagName);
                            Map<String, String> attMap = tagMap.get(tagName);
                            for (String attName : attMap.keySet()) {
                                if (target.contains(attName)) {
                                    context.setTagAttributes(attName, attMap.get(attName));
                                }
                            }
                        }
                    }
```

Reply:
I can replace it here and the List of elements has to be obtained again from a Source object built from this new replaced String. It will affect every other test. :)

Reply:
This was discussed in IRC, it will be kept as is.